python/lsst/daf/butler/core/storageClass.py

```diff
@@ -244,17 +257,22 @@ def addFromConfig(self, config):
         self._configs.append(sconfig)
 
         # Since we can not assume that we will get definitions of
-        # components before the definitions of the composites, we create
-        # two lists
+        # components or parents before the definitions of the
+        # composites/parents, we create multiple lists
+        # Children are assumed not to be themselves specifying different
+        # components for now. Ideally this should be recursive.
         composites = {}
         simple = {}
+        children = {}
         for name, info in sconfig.items():
             if "components" in info:
                 composites[name] = info
+            elif "parents" in info:
+                children[name] = info
```

Comment:
What do you think about calling this `inherits` instead of `parents` to avoid confusion between compositions and inheritance?

Reply:
Ok, although `inheritsFrom` sounds better to me.